README.md

```diff
@@ -137,14 +137,38 @@ juju relate temporal-worker-k8s admin/cos.loki
 juju relate temporal-worker-k8s admin/cos.prometheus
 ```
 
-
 ```bash
 # Access grafana with username "admin" and password:
 juju run grafana/0 -m cos get-admin-password --wait 1m
 # Grafana is listening on port 3000 of the app ip address.
 # Dashboard can be accessed under "Temporal Worker SDK Metrics", make sure to select the juju model which contains your Temporal worker operator charm.
 ```
 
+## Vault
+
+The Temporal worker operator charm can be related to the
+[Vault operator charm](https://charmhub.io/vault-k8s) to securely store
+credentials that can be accessed by workflows. This is the recommended way of
+storing workflow-related credentials in production environments. To enable this,
+run the following commands:
+
+```bash
+juju deploy vault-k8s --channel edge
+
+# After following Vault doc instructions to unseal Vault
+juju relate temporal-worker-k8s vault-k8s
+```
+
+For a reference on how to access credentials from Vault through the workflow
```

Comment:
Should there be mention here that you need to unseal the Vault, basically a link to the [tutorial](https://charmhub.io/vault-k8s/docs/h-getting-started?channel=1.15/edge).